### DISCUSSION TITLE
[Bug] Apps Pinned to Windows Taskbar Does not Open Updated App

### DISCUSSION THREAD

**Discussion by brian316** [ScoopInstaller/Scoop#5282] [Help and Questions]
#### Current Behavior

Example: 
> I have vscode version 1.73 pinned to my windows task bar. I use scoop to update the app to 1.74.
When I click the pinned icon it opens the old version 1.73 instead unless I unpin then open the application and pin again.

This happens on all apps pinned.

#### Expected Behavior

Pinned apps should always open the newest app version. That already happens if I were to install an exe program directly

#### Additional context/output

None

#### Possible Solution

Not sure. Maybe it is a symlink issue?

### System details

**Windows version:** 11 - 22H2

**OS architecture:** 64 bit

**PowerShell version:** [output of `"$($PSVersionTable.PSVersion)"`]
```
Major  Minor  Patch  PreReleaseLabel BuildLabel
-----  -----  -----  --------------- ----------
7      3      0
```

**Additional software:** N/A

#### Scoop Configuration

```json
{
  "last_update": "2022-12-08T09:12:38.2958787-08:00",
  "scoop_repo": "https://github.com/ScoopInstaller/Scoop",
  "scoop_branch": "master"
}
```

**Comment by HUMORCE** [comment]
check the TARGET propertie of the shortcut.

implementing  #5234 helps solve this problem.

**Comment by rashil2000** [comment]
How are you pinning the app on taskbar? If you pin them using the shortcuts in Start menu, the target should already contain "current".

**Reply by brian316** [reply]
I open the app through start menu. then right blick icon on bottom to pin

**Reply by rashil2000** [reply]
Can you try pinning it directly from the start menu (without opening)?

**Reply by fnagel** [reply]
> Can you try pinning it directly from the start menu (without opening)?

@rashil2000 This does not work for me. I always need to unpin and re-pin the apps the taskbar after updating them. Any idea?

**Comment by HUMORCE** [comment]
See: https://github.com/ScoopInstaller/Extras/issues/9910#issuecomment-1338797843

**Comment by elisimpson** [comment]
I'm seeing this issue too. The problem is that even when you run an app from the "current" directory (either by shortcut or directly clicking the .exe) the app *actually* runs from the versioned directory. You can see this if you open task manager and select properties from the running app, it will show it's location as the versioned directory instead of "current". This breaks the pinned taskbar icons and probably other functionality as well that expects the running app to be in the "current" directory.

Is there some way to change the way the links are creaated so the app runs from the "current" directory as expected? Not sure if they're currently symlinks, hard links, or junctions.
